### PR TITLE
Closes issue #60 and #57

### PR DESCRIPTION
**Pull request by karthikp-ubc**
Added injection capability for both src and dest registers - allreg option. Need to update the documentation in the wiki.

### CODE REVIEW

File: llvm_passes/core/Controller.cpp
Comment (abrahamchan) [inline llvm_passes/core/Controller.cpp:76]: Perhaps this is being pedantic, but could you please change this from 'injection' into 'inject'?

File: llvm_passes/core/FaultInjectionPass.cpp
Comment (abrahamchan) [inline llvm_passes/core/FaultInjectionPass.cpp:100]: Could you please place the ! closer to the expression? `!FunctionType::isValidReturnType(returntype)`

File: llvm_passes/core/FaultInjectionPass.cpp
Comment (abrahamchan) [inline llvm_passes/core/FaultInjectionPass.cpp:101]: The usual LLVM coding style is to place the || at the end of the previous line.

File: sample_programs/cpp_sample_programs/factorial/input.yaml
Comment (abrahamchan) [inline sample_programs/cpp_sample_programs/factorial/input.yaml:10]: Should we also add this option `allreg` to the master YAML file, under `docs/input_masterlist.yaml` and `docs/input_masterlist_ml.yaml` ?

**Review by abrahamchan — CHANGES_REQUESTED** [review]
Thanks for submitting this PR. Just left some minor comments.

File: sample_programs/cpp_sample_programs/factorial/input.yaml
Comment (karthikp-ubc) [reply sample_programs/cpp_sample_programs/factorial/input.yaml:10]: Yes, we should - I didn't know how to do this via a PR though. The easiest way would be to directly edit the Wiki page

**Comment by karthikp-ubc** [comment]
I've addressed the above comments. I also updated the input.yaml files for a bunch of different program to test the allreg functionality, but I'll merge it in a separate PR after this one has been merged. Let me know if there're any other comments.

File: llvm_passes/core/FaultInjectionPass.cpp
Comment (abrahamchan) [inline llvm_passes/core/FaultInjectionPass.cpp:101]: Please align the `isa<IntrinsicInst>(fi_inst)` with the same indent as `!FunctionType`. Otherwise, everything else looks good.

**Comment by karthikp-ubc** [comment]
Just addressed the comment. I'm going to merge this PR now. Thanks.

**Pull request merged**
